commit 808e277e6b130cdb3e175a19752927fe81cc464c
Author: Satish Balay <balay@mcs.anl.gov>
Date:   Mon Dec 17 15:57:19 2018 -0600

    mfem: work-around build error on ALCF/Theta with PETSc includes
    
    In file included from /gpfs/mira-home/balay/spack/opt/spack/cray-cnl6-mic_knl/intel-18.0.0.128/petsc-3.10.2-yxyele74o3vut3oparwksgs33ti2lmi5/include/petscsys.h(470),
                     from /gpfs/mira-home/balay/spack/opt/spack/cray-cnl6-mic_knl/intel-18.0.0.128/petsc-3.10.2-yxyele74o3vut3oparwksgs33ti2lmi5/include/petscbag.h(4),
                     from /gpfs/mira-home/balay/spack/opt/spack/cray-cnl6-mic_knl/intel-18.0.0.128/petsc-3.10.2-yxyele74o3vut3oparwksgs33ti2lmi5/include/petsc.h(5),
                     from linalg/petsc.hpp(23),
                     from linalg/linalg.hpp(41),
                     from linalg/sparsemat.cpp(14):
    /gpfs/mira-home/balay/spack/opt/spack/cray-cnl6-mic_knl/intel-18.0.0.128/petsc-3.10.2-yxyele74o3vut3oparwksgs33ti2lmi5/include/petscmath.h(197): error: qualified name is not allowed
      typedef complexlib::complex<double> PetscComplex;
              ^

diff --git a/var/spack/repos/builtin/packages/mfem/package.py b/var/spack/repos/builtin/packages/mfem/package.py
index d6f55b656..4216ece34 100644
--- a/var/spack/repos/builtin/packages/mfem/package.py
+++ b/var/spack/repos/builtin/packages/mfem/package.py
@@ -257,6 +257,9 @@ class Mfem(Package):
             'MFEM_USE_OPENMP=%s' % yes_no('+openmp'),
             'MFEM_USE_CONDUIT=%s' % yes_no('+conduit')]
 
+        if '+petsc' in spec:
+            spec.compiler_flags['cxxflags'].append('-DPETSC_SKIP_COMPLEX')
+
         cxxflags = spec.compiler_flags['cxxflags']
         if cxxflags:
             # The cxxflags are set by the spack c++ compiler wrapper. We also
